### PR TITLE
feat(provisioning): allow dashboard provisioning

### PR DESCRIPTION
**Pull request by talboren**

### CODE REVIEW

File: keep/api/routes/dashboard.py [hunk 1]
```diff
@@ -31,6 +34,41 @@ class DashboardResponseDTO(BaseModel):
 
 
 router = APIRouter()
+logger = logging.getLogger(__name__)
+
+
+def provision_dashboards(tenant_id: str):
+    dashboards_raw = json.loads(os.environ.get("KEEP_DASHBOARDS", "[]"))
```
Comment (shahargl) [inline keep/api/routes/dashboard.py:41]: could raise exception and Keep won't start

File: keep/api/routes/dashboard.py [hunk 2]
```diff
@@ -31,6 +34,41 @@ class DashboardResponseDTO(BaseModel):
...
+    dashboards_to_provision = [
+        DashboardCreateDTO.parse_obj(dashboard) for dashboard in dashboards_raw
+    ]
+    for dashboard in dashboards_to_provision:
+        logger.info(
+            "Provisioning Dashboard",
+            extra={"dashboard_name": dashboard.dashboard_name},
+        )
+        try:
+            create_dashboard_db(
```
Comment (shahargl) [inline keep/api/routes/dashboard.py:54]: if dashboard already exists? it will create over and over?

File: tests/test_provisioning.py [hunk 3]
```diff
@@ -221,3 +221,21 @@ def test_reprovision_provider(monkeypatch, db_session, client, test_app):
     ]
     assert len(provisioned_providers) == 1
     assert provisioned_providers[0]["type"] == "prometheus"
+
+
+@pytest.mark.parametrize(
```
Comment (shahargl) [inline tests/test_provisioning.py:226]: - maybe add a test where `KEEP_DASHBOARDS` is not a valid json?
- maybe add a test where app reloads more than once?

File: keep/api/api.py [hunk 4]
```diff
@@ -263,6 +264,8 @@ async def on_startup():
             logger.info("Providers loaded successfully")
             WorkflowStore.provision_workflows_from_directory(SINGLE_TENANT_UUID)
             logger.info("Workflows provisioned successfully")
+            provision_dashboards(SINGLE_TENANT_UUID)
```
Comment (shahargl) [inline keep/api/api.py:267]: on workflows, removing workflow from the env var will de-provision it from Keep.

Comment (shahargl) [reply keep/api/api.py:267]: can dashboard be de-provisioned?

Comment (talboren) [reply keep/api/api.py:267]: IDK. I guess we can do it if the user asks for it. For now, no.

File: keep/api/routes/dashboard.py (hunk 2 above)
Comment (talboren) [reply keep/api/routes/dashboard.py:54]: No. There's a unique constraint for the name of the dashboard.
`__table_args__ = (
        UniqueConstraint(
            "tenant_id", "dashboard_name", name="unique_dashboard_name_per_tenant"
        ),
    )`

**Review by shahargl — APPROVED** [review]
lgtm

**Pull request merged**